spec/jobs/send_inactivity_emails_job_spec.rb

```diff
@@ -0,0 +1,113 @@
+require "rails_helper"
+
+RSpec.describe SendInactivityEmailsJob, type: :job do
+  let(:programme) { create(:i_belong) }
+  let(:activity_understanding) { create(:activity) }
+  let(:programme_activity_grouping_understanding) { create(:programme_activity_grouping, programme:, required_for_completion: 1, sort_key: 2) }
+  let(:programme_activity_understanding) { create(:programme_activity, programme:, activity: activity_understanding, programme_activity_grouping: programme_activity_grouping_understanding) }
+  let(:activity_resources1) { create(:activity) }
+  let(:activity_resources2) { create(:activity) }
+  let(:activity_resources3) { create(:activity) }
+  let(:programme_activity_grouping_resources) { create(:programme_activity_grouping, programme:, required_for_completion: 3, sort_key: 3) }
+  let(:programme_activity_resources1) { create(:programme_activity, programme:, activity: activity_resources1, programme_activity_grouping: programme_activity_grouping_resources) }
+  let(:programme_activity_resources2) { create(:programme_activity, programme:, activity: activity_resources2, programme_activity_grouping: programme_activity_grouping_resources) }
+  let(:programme_activity_resources3) { create(:programme_activity, programme:, activity: activity_resources3, programme_activity_grouping: programme_activity_grouping_resources) }
+  let(:activity_engagement) { create(:activity) }
+  let(:programme_activity_grouping_engagement) { create(:programme_activity_grouping, programme:, required_for_completion: 1, sort_key: 4) }
+  let(:programme_activity_engagement) { create(:programme_activity, programme:, activity: activity_engagement, programme_activity_grouping: programme_activity_grouping_engagement) }
+
+  before do
+    programme_activity_understanding
+    programme_activity_resources1
+    programme_activity_resources2
+    programme_activity_resources3
+    programme_activity_engagement
+  end
```

Comment:
As per a comment above, this whole block could be removed if these are changed to use `let!` instead